plugins/blockvolume/blockprovider/gluster-loopback/glusterloopback.go

```diff
@@ -0,0 +1,274 @@
+	blockVols := g.BlockVolumes()
+
+	for _, blockVol := range blockVols {
+		if blockVol.Name() == name {
+			hostVol = blockVol.HostVolume()
+			break
+		}
+	}
+
+	if hostVol == "" {
+		return errors.New("block volume not found")
+	}
+
+	hostDir, err := mountHost(g, hostVol)
+	if err != nil {
+		return err
+	}
+
+	blockFileName := hostDir + "/" + name
+	stat, err := os.Stat(blockFileName)
+	if err != nil {
+		return err
+	}
+
+	err = os.Remove(blockFileName)
+	if err != nil {
+		return err
+	}
+
+	size := strconv.FormatInt(stat.Size(), 10)
+	if err = utils.ResizeBlockHostingVolume(hostVol, size); err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+			"size":  stat.Size(),
+		}).Error("error in resizing the block hosting volume")
+	}
+
+	return err
+}
+
+// GetBlockVolume gives info about a gluster block volume
+func (g *GlusterBlock) GetBlockVolume(name string) (blockprovider.BlockVolume, error) {
+	var (
+		blockVolume           blockprovider.BlockVolume
+		availableBlockVolumes = g.BlockVolumes()
+	)
+
+	//TODO: looping through all block volumes to get one block vol info is not scalable, fix it
+	for _, blockVol := range availableBlockVolumes {
+		if blockVol.Name() == name {
+			blockVolume = blockVol
+			break
+		}
+	}
+
+	if blockVolume == nil {
+		return nil, errors.New("block volume not found")
+	}
+
+	glusterBlockVol := &BlockVolume{
+		name:       blockVolume.HostVolume(),
+		hostVolume: blockVolume.Name(),
+		size:       blockVolume.Size(),
+	}
+
+	return glusterBlockVol, nil
+}
+
+// BlockVolumes returns all available gluster block volume
+func (g *GlusterBlock) BlockVolumes() []blockprovider.BlockVolume {
+	var glusterBlockVolumes = []blockprovider.BlockVolume{}
+
+	volumes, err := volume.GetVolumes(context.Background())
+	if err != nil {
+		return glusterBlockVolumes
+	}
+
+	volumes = volume.ApplyFilters(volumes, volume.BlockHosted)
+
+	for _, hostVol := range volumes {
+		hostDir, err := mountHost(g, hostVol.Name)
+		if err != nil {
+			return glusterBlockVolumes
+		}
+
+		dirent, err := ioutil.ReadDir(hostDir)
+		if err != nil {
+			return glusterBlockVolumes
+		}
+
+		for _, blockVol := range dirent {
+			glusterBlockVolumes = append(glusterBlockVolumes, &BlockVolume{name: blockVol.Name(), hostVolume: hostVol.Name, size: uint64(blockVol.Size())})
+		}
+	}
+
+	return glusterBlockVolumes
+}
+
+// ProviderName returns name of block provider
+func (g *GlusterBlock) ProviderName() string {
+	return providerName
+}
+
+// BlockVolume implements blockprovider.BlockVolume interface.
+// It holds information about a gluster-block volume
+type BlockVolume struct {
+	hosts      []string
+	iqn        string
+	username   string
+	password   string
+	hostVolume string
+	name       string
+	size       uint64
+	gbID       string
+	ha         int
+}
+
+// HostAddresses returns host addresses of a gluster block vol
+func (gv *BlockVolume) HostAddresses() []string { return gv.hosts }
```

Comment:
this functions are also not needed

Reply:
then IMO this will create an issue if we add different input structures for different provisioners, isn't it?

Reply:
Yes, So we should have kept the structures different for each provider? The structure is union of all the fields of each provider. And the caller ends up calling these functions anyways as the req structure of block vol has these fields. How to go about this?

Reply:
I think we need to refactor the code to achieve this.

Reply:
Required for this patch?

Reply:
@Madhu-1 we can take this task to define a better interface for BlockVolume to support all block providers later, for now this should be good to go.

Reply:
@Madhu-1 sure I'll do that